=== modified file 'cloudinit/distros/__init__.py'
--- a/cloudinit/distros/__init__.py	2014-11-25 19:46:10 +0000
+++ b/cloudinit/distros/__init__.py	2015-01-16 19:29:48 +0000
@@ -86,7 +86,7 @@
     def set_hostname(self, hostname, fqdn=None):
         writeable_hostname = self._select_hostname(hostname, fqdn)
         self._write_hostname(writeable_hostname, self.hostname_conf_fn)
-        self._apply_hostname(hostname)
+        self._apply_hostname(writeable_hostname)
 
     @abc.abstractmethod
     def package_command(self, cmd, args=None, pkgs=None):
@@ -160,9 +160,12 @@
             util.logexc(LOG, "Failed to non-persistently adjust the system "
                         "hostname to %s", hostname)
 
-    @abc.abstractmethod
     def _select_hostname(self, hostname, fqdn):
-        raise NotImplementedError()
+        # Prefer the short hostname over the long
+        # fully qualified domain name
+        if not hostname:
+            return fqdn
+        return hostname
 
     @staticmethod
     def expand_osfamily(family_list):

=== modified file 'cloudinit/distros/arch.py'
--- a/cloudinit/distros/arch.py	2014-09-10 18:32:37 +0000
+++ b/cloudinit/distros/arch.py	2015-01-16 19:29:48 +0000
@@ -118,13 +118,6 @@
                 return False
         return True
 
-    def _select_hostname(self, hostname, fqdn):
-        # Prefer the short hostname over the long
-        # fully qualified domain name
-        if not hostname:
-            return fqdn
-        return hostname
-
     def _write_hostname(self, your_hostname, out_fn):
         conf = None
         try:

=== modified file 'cloudinit/distros/debian.py'
--- a/cloudinit/distros/debian.py	2014-09-10 18:32:37 +0000
+++ b/cloudinit/distros/debian.py	2015-01-16 19:29:48 +0000
@@ -86,13 +86,6 @@
         else:
             return distros.Distro._bring_up_interfaces(self, device_names)
 
-    def _select_hostname(self, hostname, fqdn):
-        # Prefer the short hostname over the long
-        # fully qualified domain name
-        if not hostname:
-            return fqdn
-        return hostname
-
     def _write_hostname(self, your_hostname, out_fn):
         conf = None
         try:

=== modified file 'cloudinit/distros/freebsd.py'
--- a/cloudinit/distros/freebsd.py	2015-01-06 17:02:38 +0000
+++ b/cloudinit/distros/freebsd.py	2015-01-16 19:29:48 +0000
@@ -150,11 +150,6 @@
             return default
         return hostname
 
-    def _select_hostname(self, hostname, fqdn):
-        if not hostname:
-            return fqdn
-        return hostname
-
     def _write_hostname(self, hostname, filename):
         self.updatercconf('hostname', hostname)
 

=== modified file 'cloudinit/distros/gentoo.py'
--- a/cloudinit/distros/gentoo.py	2014-09-10 18:32:37 +0000
+++ b/cloudinit/distros/gentoo.py	2015-01-16 19:29:48 +0000
@@ -97,13 +97,6 @@
         else:
             return distros.Distro._bring_up_interfaces(self, device_names)
 
-    def _select_hostname(self, hostname, fqdn):
-        # Prefer the short hostname over the long
-        # fully qualified domain name
-        if not hostname:
-            return fqdn
-        return hostname
-
     def _write_hostname(self, your_hostname, out_fn):
         conf = None
         try:

=== modified file 'cloudinit/distros/sles.py'
--- a/cloudinit/distros/sles.py	2014-01-23 19:06:13 +0000
+++ b/cloudinit/distros/sles.py	2015-01-16 19:29:48 +0000
@@ -115,13 +115,6 @@
         conf.set_hostname(hostname)
         util.write_file(out_fn, str(conf), 0644)
 
-    def _select_hostname(self, hostname, fqdn):
-        # Prefer the short hostname over the long
-        # fully qualified domain name
-        if not hostname:
-            return fqdn
-        return hostname
-
     def _read_system_hostname(self):
         host_fn = self.hostname_conf_fn
         return (host_fn, self._read_hostname(host_fn))

